client/src/app/modules/auth/reset-password/reset-password.component.ts

```diff
@@ -58,7 +59,10 @@ export class ResetPasswordComponent implements OnInit {
 
   onSubmit(value: ConfirmPasswordDTO) {
     this.authService.resetPassword(value.password, this.token).subscribe(() => {
-      this.router.navigateByUrl('auth/sign-in');
+      this.displayConfirmDialog = true;
+      setTimeout(() => {
```

Comment:
Just a suggestion.
I'm not sure, but I think `setTimeout` is not very friendly to the angular zone and it's better to avoid it. I think we can use something from rxjs. For instance: https://stackoverflow.com/questions/70497845/alternative-to-settimeout-for-angular-lifecycles